include/core/truthTable/truth_table.hpp

```diff
@@ -0,0 +1,211 @@
+            }
+            [[nodiscard]] auto appendOne() const -> Cube {
+                return append(true);
+            }
+
+            // pass-through functions for underlying vector
+
+            auto at(std::size_t pos) -> Value& {
+                return cube[pos];
+            }
+
+            auto operator[](std::size_t pos) const -> Value {
+                return cube[pos];
+            }
+
+            auto operator<(const Cube& cv) const -> bool {
+                return (cube < cv.cube);
+            }
+
+            auto operator==(const Cube& cv) const -> bool {
+                return (cube == cv.cube);
+            }
+
+            auto reserve(const std::size_t n) -> void {
+                cube.reserve(n);
+            }
+            auto resize(const std::size_t n, const Value& val = Value()) -> void {
+                cube.resize(n, val);
+            }
+            auto emplace_back(const Value& v) -> void {
+                cube.emplace_back(v);
+            }
+
+            auto pop_back() -> void {
+                cube.pop_back();
+            }
+
+            [[nodiscard]] auto size() const -> std::size_t {
+                return cube.size();
+            }
+            [[nodiscard]] auto empty() const -> bool {
+                return cube.empty();
+            }
+            [[nodiscard]] auto begin() const -> decltype(cube.begin()) {
+                return cube.begin();
+            }
+            [[nodiscard]] auto cbegin() const -> decltype(cube.cbegin()) {
+                return cube.cbegin();
+            }
+            [[nodiscard]] auto end() const -> decltype(cube.end()) {
+                return cube.end();
+            }
+            [[nodiscard]] auto cend() const -> decltype(cube.cend()) {
+                return cube.cend();
+            }
+        };
+
+        using CubeMap = std::map<Cube, Cube>;
+
+        [[nodiscard]] auto nInputs() const -> std::size_t {
+            if (cubeMap.empty()) {
+                return 0U;
+            }
+            return cubeMap.begin()->first.size();
+        }
+
+        [[nodiscard]] auto nOutputs() const -> std::size_t {
+            if (cubeMap.empty()) {
+                return 0U;
+            }
+            return cubeMap.begin()->second.size();
+        }
+
+        auto insert(const Cube& input, const Cube& output) -> void {
+            assert(cubeMap.empty() || (input.size() == nInputs() && output.size() == nOutputs()));
+            cubeMap.try_emplace(input, output);
+        }
+        auto insert(Cube&& input, Cube&& output) -> void {
+            assert(cubeMap.empty() || (input.size() == nInputs() && output.size() == nOutputs()));
+            cubeMap.try_emplace(std::move(input), std::move(output));
+        }
+
+        auto clear() -> void {
+            cubeMap.clear();
+        }
+
+        [[nodiscard]] const CubeMap& ioCube() const {
+            return cubeMap;
+        }
+
+        auto extend() -> void;
```

Comment:
It's debatable whether this function and the `augmentWithConstants` function should be here. I'd rather consider them "algorithms" than "core functionality".